File: Perl/Rebill/cancel_recurring_payment.pl
##
# BluePay Perl Sample code.
#
# This code sample runs a $0.00 Credit Card Auth transaction
# against a customer using test payment information, sets up
# a rebilling cycle, and also shows how to cancel that rebilling cycle.
# See comments below on the details of the initial setup of the
# rebilling cycle.
##

use strict;
use lib '..';
use bluepay;

my $account_id = "Merchant's Account ID Here";
my $secret_key = "Merchant's Secret Key Here";
my $mode = "TEST";

my $rebill = BluePay->new(
    $account_id, 
    $secret_key, 
    $mode
);

$rebill->set_customer_information({
    first_name => 'Bob',
    last_name => 'Tester', 
    address1 => '1234 Test St.',
    address2 => 'Apt', 
    city => 'Testville', 
    state => 'IL', 
    zip_code => '54321',
    country => 'USA',
    phone => '123-123-12345',
    email => 'test@bluepay.com'
});

$rebill->set_cc_information({
    cc_number =>'4111111111111111', # Customer Credit Card Number
    cc_expiration => '0815', # Card Expiration Date: MMYY
    cvv2 =>'123' # Card CVV2
});

$rebill->set_recurring_payment({
    reb_first_date => "2015-01-01", # Rebill Start Date: Jan. 1, 2015
    reb_expr => "1 MONTH", # Rebill Frequency: 1 MONTH
    reb_cycles => "12", # Rebill # of Cycles: 12
    reb_amount => "15.00" # Rebill Amount: $15.00
});

$rebill->auth({amount => '0.00'}); # Card authorization amount: $0.00

# Makes the API Request with BluePay
$rebill->process();

# If transaction was approved..
if ($rebill->is_successful_response()){
    
    my $rebill_cancel = BluePay->new(
        $account_id, 
        $secret_key, 
        $mode
    );

    # Find rebill by id and cancel rebilling cycle 
    $rebill_cancel->cancel_rebilling_cycle($rebill->{REBID});

    # Makes the API request to cancel the rebill
    $rebill_cancel->process();


    # Reads the response from BluePay
    print "REBILL STATUS: " . $rebill_cancel->{status} . "\n";
    print "REBILL ID: " . $rebill_cancel->{rebill_id} . "\n";
    print "REBILL CREATION DATE: " .$rebill_cancel->{creation_date} . "\n";
    print "REBILL NEXT DATE: " .$rebill_cancel->{next_date} . "\n";
    print "REBILL LAST DATE: " .$rebill_cancel->{last_date} . "\n";
    print "REBILL SCHEDULE EXPRESSION: " .$rebill_cancel->{sched_expr} . "\n";
    print "REBILL CYCLES REMAINING: " .$rebill_cancel->{cycles_remain} . "\n";
    print "REBILL AMOUNT: " .$rebill_cancel->{reb_amount} . "\n";
    print "REBILL NEXT AMOUNT: " .$rebill_cancel->{next_amount} . "\n";
} 
else {
    print $rebill->{MESSAGE};
}

File: Python/Rebill/Cancel_Recurring_Payment.py
##
# BluePay Python Sample code.
#
# This code sample runs a $0.00 Credit Card Auth transaction
# against a customer using test payment information, sets up
# a rebilling cycle, and also shows how to cancel that rebilling cycle.
# See comments below on the details of the initial setup of the
# rebilling cycle.
##
from __future__ import print_function
import os.path, sys
sys.path.append(os.path.join(os.path.dirname(os.path.realpath(__file__)), os.pardir))
from BluePay import BluePay

account_id = "Merchant's Account ID Here"
secret_key = "Merchant's Secret Key Here"
mode = "TEST"  

payment = BluePay(
    account_id = account_id, 
    secret_key = secret_key, 
    mode = mode 
)

payment.set_customer_information(
    name1 = "Bob",
    name2 = "Tester",
    addr1 = "123 Test St.",
    addr2 = "Apt #500",
    city = "Testville",
    state = "IL",
    zipcode = "54321",
    country = "USA"
)

payment.set_cc_information(
    card_number = "4111111111111111",
    card_expire = "1215",
    cvv2 = "123"
)

# Set Recurring Payment Information
payment.set_rebilling_information(
  reb_first_date = "2015-01-05", # Rebill Start Date: Jan. 5, 2015
  reb_expr = "1 MONTH", # Rebill Frequency: 1 MONTH
  reb_cycles = "5", # Rebill # of Cycles: 5
  reb_amount = "3.50" # Rebill Amount: $3.50
)

# Auth Amount: $0.00
payment.auth(amount = '0.00')

# Makes the API Request for the recurring payment
payment.process()

# If transaction was approved..
if payment.is_successful_response():

    rebill_cancel = BluePay(
        account_id = account_id, # Merchant's Account ID
        secret_key = secret_key, # Merchant's Secret Key
        mode = mode # Transaction Mode: TEST (can also be LIVE)
    )

    # Find rebill by id and cancel rebilling cycle
    rebill_cancel.cancel_rebill(payment.reb_id_response)

    # Makes the API request to cancel the rebill
    rebill_cancel.process()

    # Read response from BluePay
    print('Rebill Status: ' + rebill_cancel.rebill_status_response)
    print('Rebill ID: ' + rebill_cancel.rebill_id_response)
    print('Rebill Creation Date: ' + rebill_cancel.creation_date_response)
    print('Rebill Next Date: ' + rebill_cancel.next_date_response)
    print('Rebill Schedule Expression: ' + rebill_cancel.sched_expression_response)
    print('Rebill Cycles Remaining: ' + rebill_cancel.cycles_remaining_response)
    print('Rebill Amount: ' + rebill_cancel.rebill_amount_response)
else:
    print(rebill_cancel.message_response)